### PR TITLE
User dropdown and notifications popover improvements

### PR DESCRIPTION
**Pull request by kimsible**
## Description

- Move the user dropdown on the avatar / username and extract the notifications popover to the right ;
**Looks more common and the old style was hard to understand for some people.**
- Improve the notifications popover appearance ;

<div style="text-align:center; display: flex; align-items: top">

<br />

</div>

- On mobile view (under 500px width), it now uses a link to /my-account/notifications instead of displaying the popover ;

- Add scrollToTop on all active links of the user dropdown (Profile, ..) and the notification popover.

## Related issues

- Fix the dropdown and popover detaching or latency jump for re-positioning effect when scrolling window or menu ;
- Fix sensitive icon hover top in user dropdown.

## Has this been tested?

- [x] 👍 yes, light tests as follows are enough

Tested on desktop view for the user dropdown and the notifications popover : 
- Make sure scrolling body and menu doesn't detach when opened ;
- Make sure the element is closed after clicking on a inside link or everywhere outside ;
- Make sure the user dropdown disapears when body scroll or and stays fixed when menu scroll ;
- Make sure the notifications popover stays fixed when body or menu scroll ;
- Make sure the active effect of the user avatar and name, and the notifications icon are well synced with dropdown or popover events ;

Tested on mobile view :
- Make sure the notifications popover doesn't display, instead a link to /my-account/notifications ;
- Make sure the user dropdown fits well the screen width and is attached to the menu when scroll. 

## Demos

On desktop : 
user-menu-demo-desktop

On mobile : 
user-menu-demo-mobile

### CODE REVIEW

**Comment by kimsible** [comment]
Do you think the notifications popover would be better with a bell as it is more common ? 

user-menu4

**Comment by test2a** [comment]
the bell icon is now so much used in the youtube universe that it has lost its meaning. i know about a toddler who told his father when he was tucking him in bed "press the bell icon and subscribe".  ;-)

i am not against the bell icon. it is now a classic icon for "notifications".  even kde uses it. 

question. why should keyboard shortcuts show in the mobile view? that is redundant with touch screens. isnt it?

**Comment by kimsible** [comment]
> the bell icon is now so much used in the youtube universe that it has lost its meaning. i know about a toddler who told his father when he was tucking him in bed "press the bell icon and subscribe". ;-)
> 
> i am not against the bell icon. it is now a classic icon for "notifications". even kde uses it.

Yes Facebook, Twitter, Mastodon, Google & YouTube use it. Discord does not, it's very close to the PeerTube Inbox btw.
Really don't know what to do with that icon.

> question. why should keyboard shortcuts show in the mobile view? that is redundant with touch screens. isnt it?

You're right, I deactivate it for mobiles but touch devices can have a keyboard too.

**Comment by test2a** [comment]
>You're right, I deactivate it for mobiles but touch devices can have a keyboard too.

there is one "shortcuts"  in the bottom of the sidebar. that would make the same thing twice in the sidebar

**Comment by kimsible** [comment]
The last commit updates icons, the bell instead of inbox-full and the tick for « Mark all as read» 

image
image

**Review by rigelk — DISMISSED** [review]
hi @kimsible, it looks good to me! I agree with your usage of the bell icon, altough as you said both icons are used in practice.

Some things though:

1. your refactoring of the profile name link doesn't support long profile names or long actor names:

Screenshot_2020-11-26 My video channels - PeerTube

2. the icon is not aligned with the rest of the block. This is nitpicking I know, but since the bell icon is a bit larger and since you added hover highlight to both (which is good!) it is very visible IMHO:

Screenshot from 2020-11-26 12-53-00

(they are not middle-aligned vertically)

**Comment by kimsible** [comment]
Thanks for review @rigelk, it's done.

**Comment by rigelk** [comment]
@kimsible thanks, looks much better!

However the left part clickability is reliant on hover to be discernable, which mobile users don’t have. I’ll talk about it with a designer friend.

**Comment by kimsible** [comment]
> @kimsible thanks, looks much better!
> 
> However the left part clickability is reliant on hover to be discernable, which mobile users don’t have. I’ll talk about it with a designer friend.

Indeed, thanks the remark =). We may simply display the grey background everytime for touch screens ?

**Comment by kimsible** [comment]
@rigelk small addition of the weekend on touchscreens, the hover effect will always be visible : 

image

**Comment by kimsible** [comment]
@rigelk same here, a UI/UX tag would be appropriate ?

**Comment by test2a** [comment]
@kimsible hi... this pr just got updated on https://peertube2.cpy.re/
but keyboard navigation has taken a major hit. if i am on user box, i enter the username and i press tab, i expect to be switched to password which actually is the case with a 2.4 website like https://tube.ouahpiti.info
user> tab>password>tab>enter

on https://peertube2.cpy.re/ right now it is like
user>tab>tab>tab>tab>tab>password>tab>tab>enter

so 5 tab in first step and  2 after password.

**Comment by rigelk** [comment]
@kimsible could you allow push by maintainers so that I can directly apply eventual fixes? I also have a patch ready that adds an arrow indicator, that complements your solution to always display the background on mobile:

Screen Shot 2020-12-09 at 16 47 43

**Comment by kimsible** [comment]
> @kimsible could you allow push by maintainers so that I can directly apply eventual fixes? I also have a patch ready that adds an arrow indicator, that complements your solution to always display the background on mobile:
> 

Very beautiful and good addition @rigelk !  ahaha I had a similar addition in progress locally but I wouldn't have done better, thanks for the help !

I click on « Allow edits and access to secret by maintainers »

image

Would you confirm you have an access now ?

**Comment by rigelk** [comment]
no, I still can’t push to your repository and branch 😢

**Comment by kimsible** [comment]
> no, I still can’t push to your repository and branch cry

How to give you access :/ ?

**Comment by kimsible** [comment]
@rigelk this is weird, were you able to push on distant branches before as you are a maintainer ?

image

**Comment by rigelk** [comment]
yep, I even did it with your last PR 😕

**Comment by rigelk** [comment]
@kimsible I uploaded my changes in https://github.com/rigelk/PeerTube/tree/feat/user-dropwdown-notifications if you want to check them

**Comment by kimsible** [comment]
> @kimsible I uploaded my changes in https://github.com/rigelk/PeerTube/tree/feat/user-dropwdown-notifications if you want to check them

@rigelk NP, I can push them if you want ?

**Comment by rigelk** [comment]
much appreciated, thanks 🙂

**Comment by kimsible** [comment]
@rigelk LGTM :+1: ! I've pushed your additions =)

File: client/src/app/menu/menu.component.scss [hunk 1]
```diff
@@ -364,4 +445,20 @@ menu {
...
+  .dropdown-item:hover, .dropdown-item:active {
+    &.settings-sensitive {
+      my-global-icon {
+        ::ng-deep {
+          svg {
+            margin-top: 0px !important;
+          }
+        }
+      }
+    }
```
Comment (rigelk) [inline client/src/app/menu/menu.component.scss:462]:
```suggestion
    &.settings-sensitive my-global-icon ::ng-deep svg {
            margin-top: 0px !important;
    }
```

File: client/src/app/menu/menu.component.scss [hunk 2]
```diff
@@ -343,6 +414,16 @@ menu {
...
     }
+
+    &.settings-sensitive {
+      my-global-icon {
+        ::ng-deep {
+          svg {
+            margin-top: 2px !important;
+          }
+        }
+      }
```
Comment (rigelk) [inline client/src/app/menu/menu.component.scss:425]:
```suggestion
      my-global-icon ::ng-deep svg {
            margin-top: 2px !important;
      }
```

**Review by rigelk — APPROVED** [review]

**Pull request merged**